Rx/src/ListReader.c

```diff
@@ -0,0 +1,163 @@
+/*******************************************************************************************************************************************************************
+ * This program recieves the data from the .py file
+ * The python sender outputs the data in the list [python data structure] format.
+ * The code below extracts the param values from the list.
+ * the output looks as follows
+ * 
+ * -----------------------------------SENDER OUTPUT --------------------------------
+ * [[['temperature 1', 81, 'check_limit'], ['ListReaderState_of_charge 1', 86, 'check_limit']], [['temperature 2', 46, 'check_limit'], ['ListReaderState_of_charge 2', 27, 'Limit OK']]]
+ * -----------------------------------SENDER OUTPUT END -----------------------------
+ * 
+ * The output can be assumed as follows
+ *      1) ... = unwanted characters
+ *      2) _   = space
+ *      3) number = number
+ *      4) , = , 
+ * 
+ * ...,_number,_...,_...,_number,_...,_...,_number,_...,_...,_number,(and so on)
+ * 
+ * The sequence ends as soon as a new line is entered. We will just iterate through the
+ * sequence until a new line (\n) is read.
+ * Once the '-' is read then there end of hte data buffer is read.
+ * 
+ * 
+ ********************************************************************************************************************************************************************/ 
+
+#include <stdio.h>              /* for scanf() */
+#include <assert.h>             /* for assert */
+#include "ListReader.h"         /* Units header */
+
+ListReader_t ListReaderState;
+
+//Test double.
+int (*scan)(const char *format, ...) = &scanf;
+
+static char ReadCharFromStdin(void)
+{
+    char ch;
+    scan("%c",&ch);    
+    return ch;
+}
+
+static void Readthestreamuntil(char par_ch)
+{
+    char ch;
+    while(1)
```

Comment:
Would have been a good place for a `do...while`, like `do {scanf...} while (ch != par_ch);`